src/main/java/org/junitpioneer/jupiter/CartesianValueArgumentsProvider.java

```diff
@@ -13,28 +13,29 @@
 import static java.util.stream.Collectors.toList;
 
 import java.lang.reflect.Array;
-import java.util.Arrays;
 import java.util.List;
 import java.util.function.Consumer;
 import java.util.stream.IntStream;
 import java.util.stream.Stream;
 
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.ArgumentsProvider;
 import org.junit.platform.commons.PreconditionViolationException;
 
 /**
  * This is basically a copy of ValueSourceArgumentsProvider,
  * except it does NOT support {@code @ParameterizedTest}.
  */
-class CartesianValueArgumentsProvider implements Consumer<CartesianValueSource> {
+class CartesianValueArgumentsProvider implements Consumer<CartesianValueSource>, ArgumentsProvider {
 
 	private Object[] arguments;
 
 	@Override
 	public void accept(CartesianValueSource source) {
 		// @formatter:off
 		List<Object> arrays =
-				// Declaration of <Object> is necessary due to a bug in Eclipse Photon.
-				Stream.<Object> of(
+				Stream.of(
```

Comment:
Did you check newer Eclipse versions work with this?

Reply:
Hm, I can't remember. I think I did, but I forgot to write it down. 🤦

Reply:
I will revert it just to be safe - but we should check it sometime.